### DISCUSSION TITLE
EXTRUDER_AUTO_FAN_SPEED

### DISCUSSION THREAD

**Discussion by wassereimer86** [mriscoc/Ender3V2S1#477] [Q&A]
Does anyone know why changing the EXTRUDER_AUTO_FAN_SPEED value for my Ender3S1-F4 doesn't work? I replaced the fan on my extruder and wanted to lower the speed of the new one. But changing the value to 205, 128 or even 20 doesn't have an effect.

**Comment by mriscoc** [comment]
Only the part-cooling fan is controllable by firmware, the extruder fan is connected directly to the power source.

**Reply by wassereimer86** [reply]
Hey @mriscoc and thank you for your reply and time. Where did you have the info from? Somebody on Printables said it works. He uses Klipper with:

```
[heater_fan hotend_fan]
pin: PC0
heater: extruder
heater_temp: 100.0
fan_speed: 1.0
```

The comments, where the person wrote it: https://www.printables.com/de/model/209504-ender-3-s1-hotend-fan-4020-replacement-bracket/comments

This is the same pin on the f1 board as used by the f4, as I looked it up in the code.

**Reply by wassereimer86** [reply]
I've installed Klipper and can confirm that it's working. So it's a Marlin issue or one with your firmware. If you need more information, let me know.

**Reply by mriscoc** [reply]
If it is controllable in klipper it must be also in Marlin, you can open an issue in the Marlin repository. S1 pin assigments were derivated from 422 and 427 boards, in that boards the extruder fan is connected with the electronics fan and both to the power source.

I check Marlin and the extruder fan is correcly defined:
https://github.com/mriscoc/Ender3V2S1/blob/1c733afd3d533959f6822341fadfc77dc88a9e72/Marlin/src/pins/stm32f1/pins_CREALITY_V24S1_301.h#L73

So, try to set `E0_AUTO_FAN_PIN` to `FAN1_PIN` in `Configuration_adv.h` file.

**Reply by wassereimer86** [reply]
That's what it is already set to.

https://github.com/mriscoc/Ender3V2S1/blob/Ender3V2S1-Released/configurations/Ender3S1-F4/Configuration_adv.h#L638

**Comment by Kaan88** [comment]
Been struggling with this myself. Even if I directly set E0_AUTO_FAN_PIN to PC0 I can't change the speed. It seems Marlin has some sanity check for pins to see if they are PWM capable but I'm not a dev and can't figure out what needs changing.

**Comment by mriscoc** [comment]
Maybe relevant:
https://github.com/mriscoc/Ender3V2S1/discussions/62
https://github.com/mriscoc/Ender3V2S1/discussions/702
https://github.com/CrealityOfficial/Ender-3S1/issues/46
